zookeeper-server/src/test/java/org/apache/zookeeper/server/persistence/FileTxnSnapLogMetricsTest.java

```diff
@@ -0,0 +1,52 @@
+package org.apache.zookeeper.server.persistence;
+
+import org.apache.zookeeper.*;
+import org.apache.zookeeper.server.ServerMetrics;
+import org.apache.zookeeper.server.SyncRequestProcessor;
+import org.apache.zookeeper.test.ClientBase;
+import org.apache.zookeeper.test.QuorumUtil;
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+import static org.hamcrest.number.OrderingComparison.greaterThan;
+import static org.hamcrest.number.OrderingComparison.greaterThanOrEqualTo;
+
+public class FileTxnSnapLogMetricsTest extends ZKTestCase {
+
+    @Test
+    public void testFileTxnSnapLogMetrics() throws Exception {
+        SyncRequestProcessor.setSnapCount(100);
+
+        QuorumUtil util = new QuorumUtil(1);
+        util.startAll();
+
+        byte[] data = new byte[500];
+        // make sure a snapshot is taken and some txns are not in a snapshot
+        ZooKeeper zk = ClientBase.createZKClient(util.getConnString());
+        for (int i=0; i<150; i++){
+            zk.create("/path" + i, data, ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
+        }
+
+        Thread.sleep(500);
```

Comment:
why do we need this sleep ?
This would be a flaky test in the future,
can't we wait for a condition ?

Reply:
fixed